src/providers/twitch/TwitchChannel.cpp

```diff
@@ -927,6 +927,69 @@ void TwitchChannel::refreshCheerEmotes()
         .execute();
 }
 
+void TwitchChannel::createClip()
+{
+    if (!this->isLive())
+    {
+        this->addMessage(makeSystemMessage(
+            "Cannot create clip while the channel is offline!"));
+        return;
+    }
+
+    this->addMessage(makeSystemMessage("Creating clip..."));
```

Comment:
In my opinion, this would be the most (second one to this being `Helix::createClip()`) convenient place to add it.

Reply:
Thanks for pointing it out and more importantly giving me an excellent example of code in action. I think there's no point in having few clips being created at once, so I guess I will go with the `QMutex`. Thanks again :heart: